File: SmartFarm/app/src/main/java/com/example/smartfarm/Scheduler.java
package com.example.smartfarm;

import java.util.UUID;

public class Scheduler {
    public String id;
    public int cycle;
    public int flow1;
    public int flow2;
    public int flow3;
    public boolean isActive;
    public String schedulerName;
    public String startTime;
    public String stopTime;

    public Scheduler(String id, int cycle, int flow1, int flow2, int flow3, boolean isActive, String schedulerName, String startTime, String stopTime) {
        this.id = id;
        this.cycle = cycle;
        this.flow1 = flow1;
        this.flow2 = flow2;
        this.flow3 = flow3;
        this.isActive = isActive;
        this.schedulerName = schedulerName;
        this.startTime = startTime;
        this.stopTime = stopTime;
    }

    // Getters and setters for all fields

    @Override
    public String toString() {
        return "Scheduler{" +
                "id='" + id + '\'' +
                ", cycle=" + cycle +
                ", flow1=" + flow1 +
                ", flow2=" + flow2 +
                ", flow3=" + flow3 +
                ", isActive=" + isActive +
                ", schedulerName='" + schedulerName + '\'' +
                ", startTime='" + startTime + '\'' +
                ", stopTime='" + stopTime + '\'' +
                '}';
    }
}

File: Scheduler.py
import datetime
import json
import time
import threading
import uuid
import asyncio
from mqtt_client import MqttClient



       
        
class Scheduler:
    def __init__(self, data):
        self.id = str(uuid.uuid4())
        self.cycle = data.get("cycle")
        self.flow1 = data.get("flow1")
        self.flow2 = data.get("flow2")
        self.flow3 = data.get("flow3")
        self.is_active = data.get("isActive")
        self.scheduler_name = data.get("schedulerName")
        self.start_time = datetime.datetime.strptime(data.get("startTime"), "%H:%M").time()
        self.stop_time = datetime.datetime.strptime(data.get("stopTime"), "%H:%M").time()

    def to_dict(self):
        return {
            "id":self.id,
            "cycle": self.cycle,
            "flow1": self.flow1,
            "flow2": self.flow2,
            "flow3": self.flow3,
            "isActive": self.is_active,
            "schedulerName": self.scheduler_name,
            "startTime": self.start_time.strftime("%H:%M"),
            "stopTime": self.stop_time.strftime("%H:%M")
        }

